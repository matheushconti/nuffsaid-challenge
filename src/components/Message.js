import React from 'react'
import { Grid, Typography, Paper, Link, makeStyles, withStyles } from '@material-ui/core'

const StyledPaper = withStyles({
  root: {
    padding: 20,
    marginBottom: 10
  },
})(Paper)

const useStyles = makeStyles((theme) => ({
  messageContainer: {
    minHeight: 70,
  },
  clearContainer: {
    display: 'flex',
    alignItems: 'flex-end'
  },
}))

export default function MessageGrid(props) {
  const classes = useStyles()
  const { message, type, removeMessage, datakey } = props

  return (
        <StyledPaper 
            style={{ backgroundColor: type.color }}
          >
            <Grid container spacing={2} alignContent="flex-end">
              <Grid sm item className={classes.messageContainer}>
                <Typography variant="body2">
                  {message.message} 
                </Typography>
              </Grid>
              <Grid item className={classes.clearContainer} >
                <Typography variant="body2">
                  <Link color="inherit" href="#" className={classes.button} onClick={() => removeMessage(datakey)}>Clear</Link> 
                </Typography>
              </Grid>
            </Grid>
          </StyledPaper>
  )

}
