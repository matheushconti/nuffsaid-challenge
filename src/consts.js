export const messageTypes = {
    error: {priority: 1, color: '#F56236', title:'Error Type 1'},
    warning: {priority: 2, color: '#FCE788', title:'Warning Type 2'},
    info: {type: 3, color: '#88FCA3', title:'Info Type 3'}
}