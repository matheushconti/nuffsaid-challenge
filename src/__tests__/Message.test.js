import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Message from '../components/Message'
import { messageTypes } from '../consts'

configure({ adapter: new Adapter() });

const messageError = {priority: 1, message: "error message"};
const messageWarning = {priority: 2, message: "warning message"};
const messageInfo = {priority: 3, message: "info message"};
const removeMessage = jest.fn();

describe('Testing Message component', () => {
    test('renders message for priority 1 (error)', () => {
        const tree = renderer.create(
            <Message 
              key={messageError.priority}
              datakey={messageError.priority}
              message={messageError.message}
              type={messageTypes.error}
              removeMessage={removeMessage}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('renders message for priority 2 (warning)', () => {
        const tree = renderer.create(
            <Message 
              key={messageWarning.priority}
              datakey={messageWarning.priority}
              message={messageWarning.message}
              type={messageTypes.warning}
              removeMessage={removeMessage}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('renders message for priority 3 (info)', () => {
        const tree = renderer.create(
            <Message 
              key={messageInfo.priority}
              datakey={messageInfo.priority}
              message={messageInfo.message}
              type={messageTypes.info}
              removeMessage={removeMessage}
            />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})