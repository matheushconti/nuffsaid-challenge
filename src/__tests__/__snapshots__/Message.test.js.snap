// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Message component renders message for priority 1 (error) 1`] = `
<div
  className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "backgroundColor": "#F56236",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
  >
    <div
      className="MuiGrid-root makeStyles-messageContainer-1 MuiGrid-item MuiGrid-grid-sm-true"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-clearContainer-2 MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Clear
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing Message component renders message for priority 2 (warning) 1`] = `
<div
  className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "backgroundColor": "#FCE788",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
  >
    <div
      className="MuiGrid-root makeStyles-messageContainer-1 MuiGrid-item MuiGrid-grid-sm-true"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-clearContainer-2 MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Clear
        </a>
      </p>
    </div>
  </div>
</div>
`;

exports[`Testing Message component renders message for priority 3 (info) 1`] = `
<div
  className="MuiPaper-root WithStyles(ForwardRef(Paper))-root-3 MuiPaper-elevation1 MuiPaper-rounded"
  style={
    Object {
      "backgroundColor": "#88FCA3",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-content-xs-flex-end"
  >
    <div
      className="MuiGrid-root makeStyles-messageContainer-1 MuiGrid-item MuiGrid-grid-sm-true"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      />
    </div>
    <div
      className="MuiGrid-root makeStyles-clearContainer-2 MuiGrid-item"
    >
      <p
        className="MuiTypography-root MuiTypography-body2"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Clear
        </a>
      </p>
    </div>
  </div>
</div>
`;
